tests/test_style.py

```diff
@@ -19,3 +21,8 @@ def test_black():
 
 def test_flake8():
     subprocess.check_call(["flake8"] + PY_FILES)
+
+
+@pytest.mark.modifies_files
+def test_pyupgrade():
+    subprocess.check_call(["pyupgrade", "--py39-plus"] + PY_FILES)
```

Comment:
I'm don't think that adding `pyupgrade` here is necessary (or useful). Feels like a better thing to do would be to run it manually when we upgrade a python version, but not necessarily to embed this in a test like this.

It also makes modifications that are arguably more than just related to Python version upgrades and are more a matter of preference or opinion.

Reply:
Interesting :thinking: 
I'm looking at this as an additional linter, like `clippy`, that guides code towards being more conventional and Pythonic.
Are there any lints in `pyupgrade` in particular that you would like to avoid?

Reply:
I think there is room for personal preference in some of their suggestions, particularly around `dict`, `set`, comprehensions, string formatting, etc... Also, the majority of the things this package covers don't appear to be very relevant to this code base.

I think there is a subset of these linting suggestions that is good (e.g. class definitions, f-strings), but I'm not sure it justifies integrating this library into the CI process, and at this scale they may be better handled on a case by case basis. I'm not too picky here, but it feels like it is more suitable for an as-needed basis whenever there is a major syntax change in a Python version bump. There's probably a reason why some of these things are not in `black` or `flake8`.